### PR TITLE
SYNCOPE-1570: Support U2F device registration via REST APIs

### PR DESCRIPTION
**Pull request by mmoayyed**
Supports U2F device registrations via dedicated REST APIs in Syncope, by connecting registration records to `AuthProfile` entries.

### CODE REVIEW

File: common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/U2FRegistrationService.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        @ApiResponse(responseCode = "201",
+            description = "U2FRegistration successfully created", headers = {
+            @Header(name = RESTHeaders.RESOURCE_KEY, schema =
+            @Schema(type = "string"),
+                description = "UUID generated for the entity created")})})
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON, RESTHeaders.APPLICATION_YAML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.APPLICATION_JSON, RESTHeaders.APPLICATION_YAML, MediaType.APPLICATION_XML})
+    @Path("devices")
+    Response save(@NotNull U2FRegisteredDevice acct);
```
Comment (ilgrosso) [inline common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/U2FRegistrationService.java:88]: Please rename this method as `save()`.

No update available?

File: common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/U2FRegistrationService.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+@SecurityRequirements({
+    @SecurityRequirement(name = "BasicAuthentication"),
+    @SecurityRequirement(name = "Bearer")})
+@Path("wa/u2f")
+public interface U2FRegistrationService extends JAXRSService {
+    @DELETE
+    @Consumes({MediaType.APPLICATION_JSON, RESTHeaders.APPLICATION_YAML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.APPLICATION_JSON, RESTHeaders.APPLICATION_YAML, MediaType.APPLICATION_XML})
+    @Path("devices")
+    Response deleteAll();
```
[the remark this answers is not in the document]
Comment (mmoayyed) [reply common/am/rest-api/src/main/java/org/apache/syncope/common/rest/api/service/wa/U2FRegistrationService.java:58]: Sounds excellent. Will do.

**Comment by ilgrosso** [comment]
@mmoayyed maybe you missed https://github.com/apache/syncope/pull/197/files#r450679988 ?

**Comment by ilgrosso** [comment]
It seems there are errors with IT:

* https://travis-ci.org/github/apache/syncope/jobs/706492140#L720
* https://travis-ci.org/github/apache/syncope/jobs/706492145#L859

**Pull request merged**